library/src/main/kotlin/it/krzeminski/githubactions/yaml/ToYaml.kt

```diff
@@ -6,19 +6,30 @@ import it.krzeminski.githubactions.domain.Workflow
 import it.krzeminski.githubactions.dsl.toBuilder
 import kotlin.io.path.invariantSeparatorsPathString
 
-fun Workflow.toYaml(addConsistencyCheck: Boolean = true): String {
+fun Workflow.toYaml(addConsistencyCheck: Boolean = true, useGitDiff: Boolean = false): String {
     val jobsWithConsistencyCheck = if (addConsistencyCheck) {
         val consistencyCheckJob = this.toBuilder().job(
             name = "check_yaml_consistency",
             runsOn = UbuntuLatest,
         ) {
             uses("Check out", CheckoutV2())
             run("Install Kotlin", "sudo snap install --classic kotlin")
-            run(
-                "Consistency check",
-                "diff -u '${targetFile.invariantSeparatorsPathString}' " +
-                    "<('${sourceFile.invariantSeparatorsPathString}')"
-            )
+            if (useGitDiff) {
+                run(
+                    "Execute script",
+                    "rm '$targetFile' && " + sourceFile.invariantSeparatorsPathString
+                )
+                run(
+                    "Consistency check",
+                    "test \$(git diff '${targetFile.invariantSeparatorsPathString}' | wc -l) -eq 0"
```

Comment:
How about simplifying it to
```suggestion
                    "git diff --exit-code '${targetFile.invariantSeparatorsPathString}'"
```

https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---exit-code

Reply:
i completely overlooked that, good find